File: python/lc-38.py
class Solution:
    def countAndSay(self, n: int) -> str:
        res = ["1"]
        for i in range(n - 1):
            newRes = []
            j = 0
            while j < len(res):
                k = j + 1
                while k < len(res) and res[k] == res[j]:
                    k += 1
                newRes.append(str(k - j))
                newRes.append(res[j])
                j = k

            res = newRes
        return "".join(res)

File: golang/lc-38.go
package main

import "strconv"

func countAndSay(n int) string {
	res := []byte{'1'}
	for i := 1; i < n; i++ {
		j := 0
		newRes := make([]byte, 0, len(res))
		for j < len(res) {
			k := j + 1
			for k < len(res) && res[k] == res[j] {
				k++
			}
			newRes = append(newRes, []byte(strconv.Itoa(k-j))...)
			newRes = append(newRes, res[j])
			j = k
		}
		res = newRes
	}
	return string(res)
}
